ipatests/test_integration/test_user_permissions.py

```diff
@@ -93,3 +93,58 @@ def test_stageuser_show_as_alternate_admin(self):
         # the field Kerberos Keys available must contain True
         result = self.master.run_command(['ipa', 'stageuser-show', stageuser])
         assert 'Kerberos keys available: True' in result.stdout_text
+
+
+class TestInstallClientNoAdmin(IntegrationTest):
+    num_clients = 1
+
+    def test_installclient_as_user_admin(self):
+        """ipa-client-install should not use hardcoded admin for principal
+        Related to : https://pagure.io/freeipa/issue/5406
+        """
+        client = self.clients[0]
+        tasks.install_master(self.master)
+        tasks.kinit_admin(self.master)
+        username = 'testuser1'
+        password = 'userSecretPassword123'
+        password_confirmation = "%s\n%s\n" % (password,
+                                              password)
+
+        self.master.run_command(['ipa', 'user-add', username,
+                                 '--first', username,
+                                 '--last', username,
+                                 '--password'],
+                                stdin_text=password_confirmation)
+
+        runcmd1 = ['ipa', 'role-add', 'useradmin']
+        self.master.run_command(runcmd1)
+        stdin_text = "useradmin\nHost Enrollment\n"
+        self.master.run_command(['ipa', 'role-add-privilege'], stdin_text)
+        runcmd3 = ['ipa', 'role-add-member', 'useradmin',
+                   '--users=' + username]
+        self.master.run_command(runcmd3)
+
+        tasks.apply_common_fixes(client)
+        tasks.allow_sync_ptr(self.master)
+        zone, error = tasks.prepare_reverse_zone(self.master, client.ip)
+        if not error:
+            self.master.run_command(["ipa", "dnszone-mod", zone,
+                                     "--dynamic-update=TRUE"])
+
+        user_kinit_stdin_text = "%s\n%s\n%s\n" % (password, password,
+                                                  password)
+
+        self.master.run_command(['kinit', username],
+                                stdin_text=user_kinit_stdin_text)
+
+        cmd = ['ipa-client-install', '-U',
+               '--domain', client.domain.name,
+               '--realm', client.domain.realm,
+               '--principal', username,
+               '-w', password,
+               '--server', self.master.hostname]
+        client.run_command(cmd)
+        msg = "getent passwd %s@%s" % (username, client.domain.name)
+        IPACLIENT_INSTALL_LOG = "/var/log/ipaclient-install.log"
+        install_log = client.get_file_contents(IPACLIENT_INSTALL_LOG)
```

Comment:
Please use ``paths.IPACLIENT_INSTALL_LOG `` from ``ipaplatform.paths`` here.

Reply:
yes before this i used that but it was failing to import that https://travis-ci.org/freeipa/freeipa/jobs/395342780#L2561

Reply:
Yeah, that's a Python 2 problem. You have to add ``from __future__ import absolute_imports`` to the top of the file.